lib/evaluate/fold.cc

```diff
@@ -554,6 +620,10 @@ Expr<Type<TypeCategory::Integer, KIND>> FoldOperation(FoldingContext &context,
     } else if (name == "rank") {
       // TODO assumed-rank dummy argument
       return Expr<T>{args[0].value().Rank()};
+    } else if (name == "reshape") {
+      if (auto maybeConstant{Reshape<T>(context, args)}) {
+        return Expr<T>{std::move(maybeConstant.value())};
```

Comment:
It's okay to use `*std::move(maybeConstant)` instead of `.value()` immediately after checking that the value is present, as is done here, if you prefer.

Reply:
`Reshape` function was changed so this part is not needed anymore